operator/src/main/java/oracle/kubernetes/operator/helpers/JobStepContext.java

```diff
@@ -0,0 +1,279 @@
+    String logHome = getLogHome();
+    if (logHome == null || "".equals(logHome.trim())) {
+      // logHome not specified, use default value
+      return DEFAULT_LOG_HOME + File.separator + getDomainUID();
+    }
+    return logHome;
+  }
+
+  String getIncludeServerOutInPodLog() {
+    return getDomain().getIncludeServerOutInPodLog();
+  }
+
+  protected String getIntrospectHome() {
+    return getDomainHome();
+  }
+
+  private ResponseStep<V1Job> createResponse(Step next) {
+    return new CreateResponseStep(next);
+  }
+
+  private class CreateResponseStep extends ResponseStep<V1Job> {
+    CreateResponseStep(Step next) {
+      super(next);
+    }
+
+    @Override
+    public NextAction onFailure(Packet packet, CallResponse<V1Job> callResponse) {
+      return super.onFailure(packet, callResponse);
+    }
+
+    @Override
+    public NextAction onSuccess(Packet packet, CallResponse<V1Job> callResponse) {
+      logJobCreated();
+      V1Job job = callResponse.getResult();
+      if (job != null) {
+        packet.put(ProcessingConstants.DOMAIN_INTROSPECTOR_JOB, job);
+      }
+      return doNext(packet);
+    }
+  }
+
+  // ---------------------- model methods ------------------------------
+
+  private V1Job createJobModel() {
+    return new V1Job()
+        .metadata(createMetadata())
+        .spec(createJobSpec(TuningParameters.getInstance()));
+  }
+
+  V1ObjectMeta createMetadata() {
+    V1ObjectMeta metadata =
+        new V1ObjectMeta()
+            .name(getJobName())
+            .namespace(getNamespace())
+            .putLabelsItem(LabelConstants.RESOURCE_VERSION_LABEL, VersionConstants.DOMAIN_V1)
+            .putLabelsItem(LabelConstants.DOMAINUID_LABEL, getDomainUID())
+            .putLabelsItem(LabelConstants.DOMAINNAME_LABEL, getDomainName())
+            .putLabelsItem(LabelConstants.CREATEDBYOPERATOR_LABEL, "true");
+    return metadata;
+  }
+
+  protected V1JobSpec createJobSpec(TuningParameters tuningParameters) {
+    V1JobSpec jobSpec =
+        new V1JobSpec().backoffLimit(0).template(createPodTemplateSpec(tuningParameters));
+
+    return jobSpec;
+  }
+
+  private V1PodTemplateSpec createPodTemplateSpec(TuningParameters tuningParameters) {
+    V1ObjectMeta metadata = new V1ObjectMeta().name(getJobName());
+    V1PodTemplateSpec podTemplateSpec =
+        new V1PodTemplateSpec().metadata(metadata).spec(createPodSpec(tuningParameters));
+    return podTemplateSpec;
+  }
+
+  private V1PodSpec createPodSpec(TuningParameters tuningParameters) {
+    V1PodSpec podSpec =
+        new V1PodSpec()
+            .activeDeadlineSeconds(60L)
+            .restartPolicy("Never")
+            .addContainersItem(createContainer(tuningParameters))
+            .addVolumesItem(new V1Volume().name(SECRETS_VOLUME).secret(getSecretsVolume()))
+            .addVolumesItem(
+                new V1Volume().name(SCRIPTS_VOLUME).configMap(getConfigMapVolumeSource()));
+
+    // TODO: where do we get image pull secret for job pod?
```

Comment:
Resolve this...  but, doesn't Job run as the operator's image?  I was going to say that we already know that the operator's image has been pulled, but... that's NOT true.  The Job/Pod could run on a different node then where the operator is running and the operator's image might not be there.  So, we need the imagePullSecrets from the operator's own Deployment.

Reply:
Right now we’re running the job in a WebLogic image (similar to managed server pod), not the operator image.  Do we need to run the job as the operator’s image?  If so, we’ll need to spend time reconfiguring the job to the operator’s Deployment and access to the appropriate k8s resources.
With WebLogic image we have access to WLST for introspection so not sure running in Operator image will work unless it has WLST deployed and configured to it.
If it makes sense to use WLS image, we may be able to set the image pull secret using something like podSpec.setImagePullSecrets(info.getDomain().getSpec().getImagePullSecrets())

Reply:
We definitely need a WL install since we need WLST & the node manager.  How is this passing on Wercker?  I thought Wercker runs pull WL directly and need the secret.   Could it be in the cache since the create-domain sample job already pulls it?

Reply:
You may be getting "lucky".  If your job gets scheduled on a node where the WebLogic image has already been pulled then you don't need the secret (default imagePullPolicy=IfNotPresent).  However, this will blow up if you are scheduled on a node where the image has not yet been pulled.